TSOClient/tso.client/UI/Controls/UIMapWaypoint.cs

```diff
@@ -0,0 +1,315 @@
+﻿using FSO.Client.Controllers;
+using FSO.Client.UI.Framework;
+using FSO.Client.UI.Model;
+using FSO.Client.UI.Panels;
+using FSO.Client.UI.Screens;
+using FSO.Common;
+using FSO.Common.DataService;
+using FSO.Common.DataService.Model;
+using FSO.Common.Rendering.Framework.IO;
+using FSO.Common.Rendering.Framework.Model;
+using FSO.Common.Utils;
+using FSO.HIT;
+using FSO.Server.DataService.Model;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Ninject;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static FSO.Client.GameContent.FileIDs;
+
+namespace FSO.Client.UI.Controls
+{
+     public class UIMapWaypoint : UIElement
+    {
+        private readonly int SizeW = 55, SizeH = 55;
+        public enum UIMapWaypointStyle
+        {
+            YouAreHere,
+            YourHouseHere
+        }
+
+        private bool AttemptRedraw = false;
+
+        private float Completion = 0;
+        private Vector2 AnimOldLoc, AnimNewLoc;
+
+        private Binding<Avatar> MyAvatar;
+        private Binding<Lot> MyLot;
+
+        public float ZOrder;
+        protected bool Is3D
```

Comment:
You can do shorthand `Is3D => FSOEnvironment.Enable3D`, though that just looks cleaner.